tests/integration/utils.py

```diff
@@ -64,30 +64,23 @@ def assert_df_equal(expected_df, got_df, ignore_cols) -> None:
             expected_df = expected_df.drop(
                 columns=ignore_cols, errors='ignore')
 
-    try:
-        if isinstance(got_df.index, pd.RangeIndex):
-            expected_df = expected_df \
-                .sort_values(by=expected_df.columns.tolist()) \
-                .reset_index(drop=True)
-        else:
-            expected_df = expected_df \
-                .sort_values(by=expected_df.columns.tolist())
-
-        if isinstance(expected_df.index, pd.RangeIndex):
-            got_df = got_df.sort_values(by=got_df.columns.tolist()) \
-                           .reset_index(drop=True)
-        else:
-            got_df = got_df.sort_values(by=got_df.columns.tolist())
-
-    except Exception:
-        sortcols = [x
-                    for x in ['namespace', 'hostname', 'ifname', 'vrf',
-                              'peer', 'prefix', 'ipAddress', 'vlan', 'macaddr']
-                    if x in expected_df.columns]
-        if sortcols:
-            expected_df = expected_df.sort_values(by=sortcols) \
-                                     .reset_index(drop=True)
-            got_df = got_df.sort_values(by=sortcols).reset_index(drop=True)
+    # Detect which columns contain lists and convert lists to string
+    expected_df = expected_df.transform(_list_columns_to_str)
+    got_df = got_df.transform(_list_columns_to_str)
+
+    if isinstance(got_df.index, pd.RangeIndex):
+        expected_df = expected_df \
+            .sort_values(by=expected_df.columns.tolist()) \
+            .reset_index(drop=True)
+    else:
+        expected_df = expected_df \
+            .sort_values(by=expected_df.columns.tolist())
+
+    if isinstance(expected_df.index, pd.RangeIndex):
+        got_df = got_df.sort_values(by=got_df.columns.tolist()) \
+                        .reset_index(drop=True)
+    else:
+        got_df = got_df.sort_values(by=got_df.columns.tolist())
```

Comment:
Don't you have other reasons why we need try/except?

Reply:
The only reason I encountered was that columns were not sortable. With this patch every column will become sortable.
I'm not saying there is no need at all for a try except, but since the current/old code only manages the unsortable column issue this patch will do the same.